src/rendering/framebufferrenderer.cpp

```diff
@@ -66,955 +66,963 @@ namespace {
+        );
+
+        // Position
+        glVertexAttribPointer(
+            0,
+            4,
+            GL_FLOAT,
+            GL_FALSE,
+            0,
+            nullptr
+        );
+        glEnableVertexAttribArray(0);
+
+        // Saves current state
+        GLint defaultFbo;
+        glGetIntegerv(GL_FRAMEBUFFER_BINDING, &defaultFbo);
+
+        glm::ivec4 viewport = OsEng.windowWrapper().viewportPixelCoordinates();
+
+        // Main framebuffer
+        GLuint pixelSizeTexture = 0;
+        GLuint pixelSizeFramebuffer = 0;
+
+        glGenTextures(1, &pixelSizeTexture);
+        glBindTexture(GL_TEXTURE_2D_MULTISAMPLE, pixelSizeTexture);
+
+        const GLsizei ONEPIXEL = 1;
+        glTexImage2DMultisample(
+            GL_TEXTURE_2D_MULTISAMPLE,
+            _nAaSamples,
+            GL_RGBA32F,
+            ONEPIXEL,
+            ONEPIXEL,
+            true
+        );
 
-        _deferredcastData[caster] = data;
+        glViewport(0, 0, ONEPIXEL, ONEPIXEL);
 
+        glGenFramebuffers(1, &pixelSizeFramebuffer);
+        glBindFramebuffer(GL_FRAMEBUFFER, pixelSizeFramebuffer);
+        glFramebufferTexture2D(
+            GL_FRAMEBUFFER,
+            GL_COLOR_ATTACHMENT0,
+            GL_TEXTURE_2D_MULTISAMPLE,
+            pixelSizeTexture,
+            0
+        );
+
+        GLenum textureBuffers[1] = { GL_COLOR_ATTACHMENT0 };
+        glDrawBuffers(1, textureBuffers);
+
+        glClearColor(0.0f, 0.0f, 0.0f, 1.0f);
+        glClear(GL_COLOR_BUFFER_BIT);
+
+        GLenum status = glCheckFramebufferStatus(GL_FRAMEBUFFER);
+        if (status != GL_FRAMEBUFFER_COMPLETE) {
+            LERROR("MSAA Sampling pattern framebuffer is not complete");
+            return;
+        }
+
+        std::unique_ptr<ghoul::opengl::ProgramObject> pixelSizeProgram = nullptr;
         try {
-            ghoul::Dictionary deferredDict = dict;
-            //deferredDict.setValue("getEntryPath", absPath(GetEntryOutsidePath));
-            _deferredcastPrograms[caster] = ghoul::opengl::ProgramObject::Build(
-                "Deferred " + std::to_string(data.id) + " raycast",
-                absPath(vsPath),
-                absPath(deferredShaderPath),
-                deferredDict);
+            pixelSizeProgram = ghoul::opengl::ProgramObject::Build(
+                "OnePixel MSAA",
+                absPath("${SHADERS}/framebuffer/pixelSizeMSAA.vert"),
+                absPath("${SHADERS}/framebuffer/pixelSizeMSAA.frag")
 
-            using IgnoreError = ghoul::opengl::ProgramObject::IgnoreError;
-            _deferredcastPrograms[caster]->setIgnoreSubroutineUniformLocationError(
-                IgnoreError::Yes
-            );
-            _deferredcastPrograms[caster]->setIgnoreUniformLocationError(
-                IgnoreError::Yes
             );
-
-            caster->initializeCachedVariables(*_deferredcastPrograms[caster]);
         }
-        catch (ghoul::RuntimeError& e) {
+        catch (const ghoul::RuntimeError& e) {
```

Comment:
No need to catch the exception. If the building fails, the next access will be to a `nullptr`

Reply:
Removed.